### ISSUE TITLE
CloudBlobDirectory.GetBlobReference(blobName) creates a new ServiceClient instead of using existing one.

### ISSUE THREAD

**Issue by serkantkaraca** [Azure/azure-storage-net#398]
CloudBlobDirectory.GetBlobReference(blobName) creates a new ServiceClient on the returned CloudBlob instead of reusing existing one on the CloudBlobDirectory. Is this intended? Since returned CloudBlob has its own ServiceClient now we end up with a new DefaultRequestOptions and lose our settings on the top level ServiceClient. Can you please recommend? Should we set DefaultRequestOptions again on the CloudBlob?

**Comment by pemari-msft** [comment]
This does seem to be unexpected. We will file a bug for this, thanks for reporting it!

**Comment by alexangas** [comment]
Not sure if this has already been fixed elsewhere but I took a look and can reproduce the issue.

The call to `CloudBlobDirectory.GetBlobReference()` creates a new CloudBlob. The constructor signature used for this call does not supply the ServiceClient (only its Credentials property), so that is therefore not set in the new CloudBlob object.

When said constructor then calls `ParseQueryAndVerify()` it reaches [this code](https://github.com/Azure/azure-storage-net/blob/bdeb107c5fc1ad6db84fe6deaff17b15717aac0c/Lib/Common/Blob/CloudBlob.Common.cs#L492) which creates a new ServiceClient:

```csharp
if (this.ServiceClient == null)
{
    this.ServiceClient = new CloudBlobClient(NavigationHelper.GetServiceClientBaseAddress(this.StorageUri, null /* usePathStyleUris */), credentials ?? parsedCredentials);
}
```

One possible solution is to call a different CloudBlob constructor but the only signatures that take a ServiceClient or related object as parameter are marked internal. As `GetBlobReference()` is marked **public virtual**, that could cause problems for those that wish to override it.
